### PR TITLE
Add blockchain tweets pagination

### PR DESCRIPTION
**Pull request by smyzin**
- added a new function to Post store
- created a new hook for pagination
- updated `safeGetPostsFromContract` to use `skip/limit`

### CODE REVIEW

File: src/components/BlockchainList/PostsList.tsx [hunk 1]
```diff
@@ -19,44 +20,42 @@ const scrollContainer = classnames(
 )
 
 function BlockchainPostsListSuspended() {
-  const { selectedPosts } = useSnapshot(PostStore)
+  const { selectedPosts, postsAmount } = useSnapshot(PostStore)
```
Comment (ohpaul) [inline src/components/BlockchainList/PostsList.tsx:23]: suggest naming to `postAmount` instead

File: src/components/BlockchainList/PostsList.tsx [hunk 2]
```diff
@@ -19,44 +20,42 @@ const scrollContainer = classnames(
...
   const matchStore = hashStore && hashStore === selectedType
 
-  const sliceToSpecificPost = (totalPosts: number) => {
-    if (!(matchStore || hashId)) return 10
-    return totalPosts - Number(hashId)
+  const slideToSpecificPost = (totalPosts: number) => {
+    if (!(matchStore || hashId)) return PostStore.postsLimit
+    const neededLimit = totalPosts - Number(hashId)
+    return matchStore
+      ? neededLimit > PostStore.postsLimit
```
Comment (ohpaul) [inline src/components/BlockchainList/PostsList.tsx:33]: Maybe replace this with `Math.max(neededLimit, PostStore.postLimit`?

**Review by ohpaul — APPROVED** [review]

File: src/components/BlockchainList/PostsList.tsx [hunk 3]
```diff
@@ -19,44 +21,56 @@ const scrollContainer = classnames(
...
+    function updateLimitIfNeeded() {
+      if (!(matchStore || hashId)) return
+      const neededLimit = totalPosts - Number(hashId)
+      setScrolledLimit(Math.max(neededLimit, PostStore.limit))
+      window.scrollTo({ top: window.innerHeight, behavior: 'smooth' })
+    }
 
-  const posts = PostStore.selectedToken
+    void updateLimitIfNeeded()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (T-Damer) [inline src/components/BlockchainList/PostsList.tsx:41]: Hm, what if `params` will change or `limit` will change somehow? Maybe we shouldn't silence it? Maybe we can handle dependency array somehow? Also, how about to put it into helper?

Comment (T-Damer) [reply src/components/BlockchainList/PostsList.tsx:41]: Seems like a part of `scrollToAnchor`

File: src/helpers/safeGetPostsFromContract.ts [hunk 4]
```diff
@@ -15,8 +22,8 @@ export default async function (contract: SCPostStorage) {
           } as PostStructOutput)
       )
       .reverse()
-    // Reverse is needed to get the latest posts first
-  } catch {
+  } catch (error) {
+    console.error('Error occurred on getting posts error:', error)
```
Comment (T-Damer) [inline src/helpers/safeGetPostsFromContract.ts:26]: How about to send it into `handleError('Error occurred on getting posts error:', error)`?

File: src/helpers/safeGetPostsAmountFromContract.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { SCPostStorage } from '@big-whale-labs/seal-cred-posts-contract'
+
+export default async function (contract: SCPostStorage) {
+  try {
+    return Number(await contract.currentPostId())
+  } catch (error) {
+    console.error('Error occurred while running "currentPostId()"', error)
```
Comment (T-Damer) [inline src/helpers/safeGetPostsAmountFromContract.ts:7]: Let's send it into `handleError('Error occurred while getting current post id from the contract', error)`, note the message changes, I think it sounds better

File: src/helpers/getMorePosts.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    if (totalAmount === 0) return Promise.resolve([])
+
+    const leftRecords = totalAmount - loadedPostAmount
+    const finalLimit = Math.min(limitAmount, leftRecords)
+    const shouldSkip = totalAmount - finalLimit - loadedPostAmount
+    const finalSkip = Math.max(0, shouldSkip)
+
+    return safeGetPostsFromContract(contract, finalSkip, finalLimit)
+  } catch (error) {
+    console.error(error)
```
Comment (T-Damer) [inline src/helpers/getMorePosts.ts:26]: Let's do `handleError`

File: src/helpers/safeGetPostsAmountFromContract.ts (hunk 5 above)
Comment (T-Damer) [reply src/helpers/safeGetPostsAmountFromContract.ts:7]: Also, looks like you have nested `try-catch` blocks, not sure if should do this

File: src/components/BlockchainList/PostsList.tsx [hunk 7]
```diff
@@ -19,44 +21,56 @@ const scrollContainer = classnames(
...
+          ...selectedPosts,
+          ...newPosts,
+        ])
+        setScrolledLimit(PostStore.limit)
+      }}
       className={scrollContainer}
       style={{ overflow: 'hidden' }}
-      dataLength={loadedItemsAmount}
-      hasMore={loadedItemsAmount < posts.length}
+      dataLength={selectedPosts.length}
```
Comment (T-Damer) [inline src/components/BlockchainList/PostsList.tsx:69]: You're using `selectedPosts.length` a couple of times, I think it's worth to extract it into `const` or destructure

**Pull request merged**